File: 2-String/434_countSegments.py
import re
class Solution:
    def countSegments(self, s):
        """
        :type s: str
        :rtype: int
        """
        # way 1
        return sum(s[i] != ' ' and (i == 0 or s[i-1] == ' ') for i in range(len(s)))
        # way 2
        # return len(s.split(' '))
    

sol = Solution()
s = "h h "
ans = sol.countSegments(s)
print(ans)

File: 2-String/434_countSegments.java
import static java.lang.System.out;
import java.util.*;
class Solution {
    public int countSegments(String s) {
        int res=0;
        for(int i=0; i<s.length(); i++)
            if(s.charAt(i)!=' ' && (i==0 || s.charAt(i-1)==' '))
                res++;        
        return res;
    }
}

// Time complexity:  O(n)
// Space complexity: O(1)
public class Segments{
    public static void main(String[] args){
        String s = "h h ";
        Solution sol = new Solution();
        int ans = sol.countSegments(s);
        System.out.print(ans);
    }
}